src/topology/algebra/semigroup.lean

```diff
@@ -25,50 +25,45 @@ lemma exists_idempotent_of_compact_t2_of_continuous_mul_left {M} [nonempty M] [s
 begin
 /- We apply Zorn's lemma to the poset of nonempty closed subsemigroups of `M`. It will turn out that
 any minimal element is `{m}` for an idempotent `m : M`. -/
-  let S : set (set M) := { N : set M | is_closed N ∧ N.nonempty ∧ ∀ m m' ∈ N, m * m' ∈ N },
+  let S := {N : set M | is_closed N ∧ N.nonempty ∧ ∀ m m' ∈ N, m * m' ∈ N},
   suffices : ∃ N ∈ S, ∀ N' ∈ S, N' ⊆ N → N' = N,
   { rcases this with ⟨N, ⟨N_closed, ⟨m, hm⟩, N_mul⟩, N_minimal⟩,
     use m,
 /- We now have an element `m : M` of a minimal subsemigroup `N`, and want to show `m + m = m`.
 We first show that every element of `N` is of the form `m' + m`.-/
     have scaling_eq_self : (* m) '' N = N,
     { apply N_minimal,
-      { refine ⟨(continuous_mul_left m).is_closed_map _ N_closed,
-          ⟨_, ⟨m, hm, rfl⟩⟩, _⟩,
+      { refine ⟨(continuous_mul_left m).is_closed_map _ N_closed, ⟨_, ⟨m, hm, rfl⟩⟩, _⟩,
         rintros _ ⟨m'', hm'', rfl⟩ _ ⟨m', hm', rfl⟩,
-        refine ⟨m'' * m * m', N_mul _ (N_mul _ hm'' _ hm) _ hm', mul_assoc _ _ _⟩, },
+        refine ⟨_, N_mul _ (N_mul _ hm'' _ hm) _ hm', mul_assoc _ _ _⟩ },
       { rintros _ ⟨m', hm', rfl⟩,
-        exact N_mul _ hm' _ hm, }, },
+        exact N_mul _ hm' _ hm } },
 /- In particular, this means that `m' * m = m` for some `m'`. We now use minimality again to show
-that this holds for _all_ `m' ∈ N`. -/
-    have absorbing_eq_self : N ∩ { m' | m' * m = m} = N,
+that this holds for all `m' ∈ N`. -/
+    have absorbing_eq_self : N ∩ {m' | m' * m = m} = N,
     { apply N_minimal,
       { refine ⟨N_closed.inter ((t1_space.t1 m).preimage (continuous_mul_left m)), _, _⟩,
-        { rw ←scaling_eq_self at hm, exact hm },
+        { rwa ←scaling_eq_self at hm },
         { rintros m'' ⟨mem'', eq'' : _ = m⟩ m' ⟨mem', eq' : _ = m⟩,
           refine ⟨N_mul _ mem'' _ mem', _⟩,
-          rw [set.mem_set_of_eq, mul_assoc, eq', eq''], }, },
-      apply set.inter_subset_left, },
+          rw [set.mem_set_of_eq, mul_assoc, eq', eq''] } },
+      apply set.inter_subset_left },
 /- Thus `m * m = m` as desired. -/
     rw ←absorbing_eq_self at hm,
-    exact hm.2, },
-  apply zorn_superset,
-  intros c hcs hc,
-  refine ⟨⋂₀ c, ⟨is_closed_sInter $ λ t ht, (hcs ht).1, _, _⟩, _⟩,
+    exact hm.2 },
+  refine zorn_superset _ (λ c hcs hc, ⟨_, ⟨is_closed_sInter $ λ t ht, (hcs ht).1, _,
+    λ m hm m' hm', _⟩, λ s hs, set.sInter_subset_of_mem hs⟩),
```

Comment:
Same here: I'd leave an explicit `⋂₀ c`. Also, if you split it back into two steps, then a reader can examine the proof state after application of `zorn_superset`.
```suggestion
  refine zorn_superset _ (λ c hcs hc, _),
  refine ⟨⋂₀ c, ⟨is_closed_sInter $ λ t ht, (hcs ht).1, _,
    λ m hm m' hm', _⟩, λ s hs, set.sInter_subset_of_mem hs⟩),
```